cosmo_tester/test_suites/test_manager_upgrade/manager_upgrade_test.py

```diff
@@ -0,0 +1,322 @@
+########
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import json
+import os
+import shutil
+import tempfile
+import urllib2
+
+from cloudify.workflows import local
+
+from cosmo_tester.framework.testenv import TestCase
+from cosmo_tester.framework.git_helper import clone
+from cosmo_tester.framework.cfy_helper import CfyHelper
+from cosmo_tester.framework.testenv import (initialize_without_bootstrap,
+                                            clear_environment)
+from cosmo_tester.framework.util import create_rest_client, YamlPatcher
+
+from influxdb import InfluxDBClient
+
+
+def setUp():
+    initialize_without_bootstrap()
+
+
+def tearDown():
+    clear_environment()
+
+
+BOOTSTRAP_REPO_URL = 'https://github.com/cloudify-cosmo/'\
+                     'cloudify-manager-blueprints.git'
+BOOTSTRAP_BRANCH = '3.4m5'
+
+UPGRADE_REPO_URL = 'https://github.com/codilime/' \
+                   'cloudify-manager-blueprints.git'
+UPGRADE_BRANCH = 'fixup-upgrade'
+
+
+class ManagerUpgradeTest(TestCase):
+
+    # TODO check if we want to separate openstack and base testcases
+    # so that we can also have an aws-based test etc
+
+    def test_manager_upgrade(self):
+        self.prepare_manager()
+
+        self.post_bootstrap_checks()
+
+        preupgrade_deployment_id = self.deploy_hello_world('pre-')
+        upgrade_blueprint_path = self.get_upgrade_blueprint()
+
+        self.upgrade_manager(upgrade_blueprint_path)
+
+        postupgrade_deployment_id = self.deploy_hello_world('post-')
+        self.check_influx(postupgrade_deployment_id)
+        self.uninstall_deployment(postupgrade_deployment_id)
+
+        self.post_upgrade_checks(preupgrade_deployment_id)
+        self.check_influx(preupgrade_deployment_id)
+
+        self.rollback_manager(upgrade_blueprint_path)
+        self.check_influx(preupgrade_deployment_id)
+        self.uninstall_deployment(preupgrade_deployment_id)
+
+        self.post_rollback_checks()
+
+        self.teardown_manager()
+
+    @property
+    def _use_existing(self):
+        return 'upgrade_manager_ip' in self.env.handler_configuration
+
+    def make_cfy(self, workdir=None):
+        # TODO: perhaps need to use a separate cfy checkout? so cli version
+        # is the same as the manager version (pre/post-ugprade)
+        # TODO: or maybe need a --skip-version-check
```

Comment:
The CLI actually uses the skip-version-check when initializing the rest service it uses..